pcdsdevices/lodcm.py

```diff
@@ -1139,7 +1148,124 @@ def calc_energy(self, energy, material=None, reflection=None):
         material = material or self.get_material()
         reflection = reflection or self.get_reflection(
             as_tuple=True, check=True)
-        return self.energy.calc_energy(energy, material, reflection)
+        return self.energy.calc_geometry(energy, material, reflection)
+
+    def set_energy(self, energy, material=None, reflection=None):
+        """
+        Set the current positions of th1, th2, z1 and z2 offset motors.
+
+        Parameters
+        ----------
+        energy : number
+            Energy in keV.
+        material : str, optional
+            Crystal material. E.g.: `Si`
+        reflection : tuple, optional
+            Crystal's reflections. E.g.: `(1,1,1)`
+        """
+        # try to determine the material and reflection:
+        material = material or self.get_material()
+        reflection = reflection or self.get_reflection(
+            as_tuple=True, check=True)
+        th, z = diffraction.get_lom_geometry(energy * 1e3, material,
+                                             reflection)
+        # TODO: probably not the correct approach here to use
+        # set_current_position
+        if material == 'Si':
+            self.th1Si.set_current_position(th)
```

Comment:
I am not sure if I am supposed to use `set_current_position` directly here or if I should be using something else instead?

In the old code they have their own `set` method for it:

```
        def set(value):
            if use_ims_preset:
                Pv.put("%s_SET" % offset_pv, motor.wm() - value)
            Pv.put(offset_pv, motor.wm() - value)
```

And use it here:
```
        if ID == "Si":
            self.th1Si.set(th)
            self.z1Si.set(-z)
        elif ID == "C":
            self.th1C.set(th)
            self.z1C.set(-z)
```

Reply:
I completely overlooked this, thank you for pointing that out. I added a new `offset_set_pv` to the `OffsetMotorBase` in 4d32fc3 but I am not sure if that was the right approach - I think we want to only add it for certain motors not for all of them, even if we won't set those PVs for the other motors, and I am not entirely sure how I'd do that (i am going to give it more more though tomorrow):

https://github.com/pcdshub/pcdsdevices/blob/4a3e4f061557a5925c23496a7fdfedb8a3f54366/pcdsdevices/pseudopos.py#L1010-L1012

Reply:
I think we need to have a subclass of the offset motor that has this additional `{self._prefix}_SET` PV and also puts to it during `set_current_position`, then carefully pick which ones to grant this subclass to... I still don't know exactly why the old classes do this but I'm scared to upset the balance of "years of legacy code"

Reply:
Ohh that makes more sense, I will try that instead, thank you.

Reply:
I added a new subclass in f11f91e - I think this is what you meant, please let me know if I missed your suggestion.

As of now, I am not using any motors with the old `use_ims_preset=True` in any of the calculations, but the old `lodcm` object is so big - I am sure we'll have to add more stuff into it later on... and probably scientists are using these motors independently too... 

I also have no idea what is the `_SET` for, @ZryletTC  do you know?

Reply:
The `_SET` PVs are just the setpoints for certain states. In this case, I think this would be the diamond or silicon states setpoints. And so when you update the offset for a motor, the states should get changed too.
@ZLLentz does that match your understanding?